torq.q

```diff
@@ -400,6 +400,9 @@ readprocfile:{[file]
 		// map hostnames in res to order of preference, select most preferred
 		first res iasc prefs?res[`host];
 		first res];
+	// exit if no port passed via command line or specified in process.csv
+	if[null[output`port]&0i=system"p";
```

Comment:
Under the strict condition of requiring hostname to match this additional check may be enough:
```suggestion
	if[(null[output`port]|not output[`host]in prefs)&0i=system"p";
```
At least enough for (1) + (2) I reckon.